Fix default Docker registry prefix generation for `make push-manifest`

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->


**What happened**:

Using `make push-manifest` to publish private images to my personal Docker Hub repository fails with the following error:

```
$ export REGISTRY=ejreed
$ export IMAGE_VERSION=latest
$ make push-manifest
docker manifest create --amend ejreed/azuredisk-csi:latest  ejreed/azuredisk-csi:latest-linux-amd64  ejreed/azuredisk-csi:latest-windows-1809-amd64  ejreed/azuredisk-csi:latest-windows-1903-amd64  ejreed/azuredisk-csi:latest-windows-1909-amd64  ejreed/azuredisk-csi:latest-windows-2004-amd64
Created manifest list docker.io/ejreed/azuredisk-csi:latest
# add "os.version" field to windows images (based on https://github.com/kubernetes/kubernetes/blob/master/build/pause/Makefile)
set -x; \
registry_prefix=; \
manifest_image_folder=`echo "${registry_prefix}ejreed/azuredisk-csi:latest" | sed "s|/|_|g" | sed "s/:/-/"`; \
for arch in amd64; do \
	for osversion in 1809 1903 1909 2004; do \
		BASEIMAGE=mcr.microsoft.com/windows/nanoserver:${osversion}; \
		full_version=`docker manifest inspect ${BASEIMAGE} | jq -r '.manifests[0].platform["os.version"]'`; \
		sed -i -r "s/(\"os\"\:\"windows\")/\0,\"os.version\":\"${full_version}\"/" "/home/edreed/.docker/manifests/${manifest_image_folder}/${manifest_image_folder}-windows-${osversion}-${arch}"; \
	done; \
done
+ registry_prefix=
+ echo ejreed/azuredisk-csi:latest
+ sed s|/|_|g
+ sed s/:/-/
+ manifest_image_folder=ejreed_azuredisk-csi-latest
+ BASEIMAGE=mcr.microsoft.com/windows/nanoserver:1809
+ docker manifest inspect mcr.microsoft.com/windows/nanoserver:1809
+ jq -r .manifests[0].platform["os.version"]
+ full_version=10.0.17763.1637
+ sed -i -r s/("os"\:"windows")/\0,"os.version":"10.0.17763.1637"/ /home/edreed/.docker/manifests/ejreed_azuredisk-csi-latest/ejreed_azuredisk-csi-latest-windows-1809-amd64
sed: can't read /home/edreed/.docker/manifests/ejreed_azuredisk-csi-latest/ejreed_azuredisk-csi-latest-windows-1809-amd64: No such file or directory
+ BASEIMAGE=mcr.microsoft.com/windows/nanoserver:1903
+ docker manifest inspect mcr.microsoft.com/windows/nanoserver:1903
+ jq -r .manifests[0].platform["os.version"]
+ full_version=10.0.18362.1256
+ sed -i -r s/("os"\:"windows")/\0,"os.version":"10.0.18362.1256"/ /home/edreed/.docker/manifests/ejreed_azuredisk-csi-latest/ejreed_azuredisk-csi-latest-windows-1903-amd64
sed: can't read /home/edreed/.docker/manifests/ejreed_azuredisk-csi-latest/ejreed_azuredisk-csi-latest-windows-1903-amd64: No such file or directory
+ BASEIMAGE=mcr.microsoft.com/windows/nanoserver:1909
+ docker manifest inspect mcr.microsoft.com/windows/nanoserver:1909
+ jq -r .manifests[0].platform["os.version"]
+ full_version=10.0.18363.1256
+ sed -i -r s/("os"\:"windows")/\0,"os.version":"10.0.18363.1256"/ /home/edreed/.docker/manifests/ejreed_azuredisk-csi-latest/ejreed_azuredisk-csi-latest-windows-1909-amd64
sed: can't read /home/edreed/.docker/manifests/ejreed_azuredisk-csi-latest/ejreed_azuredisk-csi-latest-windows-1909-amd64: No such file or directory
+ BASEIMAGE=mcr.microsoft.com/windows/nanoserver:2004
+ docker manifest inspect mcr.microsoft.com/windows/nanoserver:2004
+ jq -r .manifests[0].platform["os.version"]
+ full_version=10.0.19041.685
+ sed -i -r s/("os"\:"windows")/\0,"os.version":"10.0.19041.685"/ /home/edreed/.docker/manifests/ejreed_azuredisk-csi-latest/ejreed_azuredisk-csi-latest-windows-2004-amd64
sed: can't read /home/edreed/.docker/manifests/ejreed_azuredisk-csi-latest/ejreed_azuredisk-csi-latest-windows-2004-amd64: No such file or directory
make: *** [Makefile:152: push-manifest] Error 2
```

**What you expected to happen**:

It should successfully generate the manifest and push to the target image label.

**How to reproduce it**:

See example above.

**Anything else we need to know?**:

**Environment**:
- CSI Driver version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/Makefile
+++ b/Makefile
@@ -150,7 +150,7 @@ push-manifest:
 	docker manifest create --amend $(IMAGE_TAG) $(foreach osarch, $(ALL_OS_ARCH), $(IMAGE_TAG)-${osarch})
 	# add "os.version" field to windows images (based on https://github.com/kubernetes/kubernetes/blob/master/build/pause/Makefile)
 	set -x; \
-	registry_prefix=$(shell (echo ${REGISTRY} | grep -Eq ".*\/.*") && echo "docker.io/" || echo ""); \
+	registry_prefix=$(shell (echo ${REGISTRY} | grep -Eq ".*[\/\.].*") && echo "" || echo "docker.io/"); \
 	manifest_image_folder=`echo "$${registry_prefix}${IMAGE_TAG}" | sed "s|/|_|g" | sed "s/:/-/"`; \
 	for arch in $(ALL_ARCH.windows); do \
 		for osversion in $(ALL_OSVERSIONS.windows); do \

